service.go

```diff
@@ -141,6 +142,26 @@ func (s *service) createHTTPComponent() (Component, error) {
 
 	b := http.NewBuilder().WithPort(int(portVal))
 
+	httpReadTimeout, ok := os.LookupEnv("PATRON_HTTP_READ_TIMEOUT_SECONDS")
+	if ok {
+		readTimeout, err := strconv.ParseInt(httpReadTimeout, 10, 64)
```

Comment:
We could use the ability of go to parse duration with https://golang.org/pkg/time/#ParseDuration.
This way we support any timescale and are not restricted with only seconds, which is too coarse-grained. What do you think? Makes parsing even simpler.

Reply:
+1 to this suggestion.  
strconv.ParseInt accepts values such as `+45` or `-100`, which would need further conditions to be handled correctly.

Reply:
> We could use the ability of go to parse duration with https://golang.org/pkg/time/#ParseDuration.
> This way we support any timescale and are not restricted with only seconds, which is too coarse-grained. What do you think? Makes parsing even simpler.

changes applied

Reply:
> +1 to this suggestion.
> strconv.ParseInt accepts values such as `+45` or `-100`, which would need further conditions to be handled correctly.

I am not sure what you mean because by using `WithReadTimeout()` and `WithWriteTimeout()`, these conditions are already handled.

```go
// WithReadTimeout sets the Read Timeout for the HTTP component.
func (cb *Builder) WithReadTimeout(rt time.Duration) *Builder {
	if rt <= 0*time.Second {
		cb.errors = append(cb.errors, errors.New("negative or zero read timeout provided"))
	} else {
		log.Infof("setting read timeout")
		cb.httpReadTimeout = rt
	}

	return cb
}

// WithWriteTimeout sets the Write Timeout for the HTTP component.
func (cb *Builder) WithWriteTimeout(wt time.Duration) *Builder {
	if wt <= 0*time.Second {
		cb.errors = append(cb.errors, errors.New("negative or zero write timeout provided"))
	} else {
		log.Infof("setting write timeout")
		cb.httpWriteTimeout = wt
	}

	return cb
}
```

Reply:
You're right!